scripts/zkai_native_attention_mlp_rmsnorm_input_fused_adapter_gate.py

```
@@ -0,0 +1,544 @@
+            fieldnames=TSV_COLUMNS,
+            dialect="excel-tab",
+            lineterminator="\n",
+        )
+        writer.writeheader()
+        writer.writerow(row)
+        write_text_atomically(tsv_path, buffer.getvalue())
+
+
+def write_text_atomically(path: pathlib.Path, text: str) -> None:
+    if path.is_symlink():
+        raise RmsnormInputFusedAdapterGateError(f"refusing to overwrite symlink: {path}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)